### PR TITLE
Load the world from filename defined in the config. 

### PR DESCRIPTION
**Pull request by 0101coding**
This is to solve the second issue mentioned in this PR #9

### CODE REVIEW

File: crates/logic/src/server/network/login.rs [hunk 1]
```diff
@@ -54,7 +54,7 @@ impl FalconServer {
             let player_abilities = PlayerAbilitiesSpec::new(player.ability_flags(), 0.05, 0.1);
             player.connection().build_send_packet(player_abilities, falcon_send::send_player_abilities);
             
-            self.world.send_chunks_for_player(player);
+            self.world.as_mut().unwrap().send_chunks_for_player(player);
```
Comment (GrizzlT) [inline crates/logic/src/server/network/login.rs:57]: `unwrap` in this case would crash the server when no world is specified, right?

File: crates/logic/src/server/network/play.rs [hunk 2]
```diff
@@ -49,7 +49,7 @@ impl FalconServer {
             None => (0, 0, 0, 0),
         };
         if update_position {
-            self.world.update_player_pos(self.players.get(&uuid).unwrap(), old_x, old_z, x, z);
+            self.world.as_mut().unwrap().update_player_pos(self.players.get(&uuid).unwrap(), old_x, old_z, x, z);
```
Comment (GrizzlT) [inline crates/logic/src/server/network/play.rs:52]: same here

File: crates/logic/src/server/network/play.rs [hunk 3]
```diff
@@ -62,7 +62,7 @@ impl FalconServer {
 
     pub fn player_update_view_distance(&mut self, uuid: Uuid, view_distance: u8) {
         if let Some(player) = self.players.get_mut(&uuid) {
-            self.world.update_view_distance(player, view_distance);
+            self.world.as_mut().unwrap().update_view_distance(player, view_distance);
```
Comment (GrizzlT) [inline crates/logic/src/server/network/play.rs:65]: and here

File: crates/main/src/server/mod.rs [hunk 4]
```diff
@@ -18,26 +19,30 @@ pub mod console;
 
 pub(crate) fn start_server(shutdown_handle: ShutdownHandle) -> Result<()> {
     info!("Starting server thread...");
-
-    let world = {
```
Comment (GrizzlT) [inline crates/main/src/server/mod.rs:22]: This should become a match. (see my review)

**Review by GrizzlT — CHANGES_REQUESTED** [review]
Thank you for this pr.

I don't think this approach would solve the issue completely given the use of `unwrap()` which would result in a crash when no world is specified.

Instead, `FalconWorld` is designed to have a dynamic amount of chunks, thus that could also be zero chunks. I think this is the more ideal way of solving the issue: if the config setting is empty, just load a `FalconWorld` with zero chunks.

**Comment by 0101coding** [comment]
Ok. Thanks. I will get to that. I thought as much

**Comment by taylorsk** [comment]
This was fixed in PR #9

**Pull request closed without merging**